src/client/pfs.go

```diff
@@ -1035,19 +1039,29 @@ type putFileWriteCloser struct {
 	sent          bool
 }
 
-func (c APIClient) newPutFileWriteCloser(repoName string, commitID string, path string, delimiter pfs.Delimiter, targetFileDatums int64, targetFileBytes int64, overwriteIndex *pfs.OverwriteIndex) (*putFileWriteCloser, error) {
+func (c APIClient) newPutFileWriteCloser(repoName string, commitID string, path string, delimiter pfs.Delimiter, targetFileDatums int64, targetFileBytes int64, overwriteIndex *pfs.OverwriteIndex, header []byte, footer []byte) (*putFileWriteCloser, error) {
 	putFileClient, err := c.PfsAPIClient.PutFile(c.Ctx())
 	if err != nil {
 		return nil, grpcutil.ScrubGRPC(err)
 	}
+	var headerValue, footerValue *pfs.Metadata
+	if header != nil {
+		headerValue = &pfs.Metadata{Value: header}
+	}
+	if footer != nil {
+		footerValue = &pfs.Metadata{Value: footer}
+	}
+	request := &pfs.PutFileRequest{
+		File:             NewFile(repoName, commitID, path),
+		Delimiter:        delimiter,
+		TargetFileDatums: targetFileDatums,
+		TargetFileBytes:  targetFileBytes,
+		OverwriteIndex:   overwriteIndex,
+		Header:           headerValue,
+		Footer:           footerValue,
+	}
```

Comment:
Could we keep this struct inlined in the return value? I find that a bit more readable.